src/storage/btree.c

```diff
@@ -14089,10 +14102,18 @@ btree_reflect_global_unique_statistics (THREAD_ENTRY * thread_p, GLOBAL_UNIQUE_S
 
       if (!only_active_tran || logtb_is_current_active (thread_p))
 	{
+	  int over = 0;
+
 	  /* update header information */
-	  root_header->num_nulls = unique_stat_info->unique_stats.num_nulls;
-	  root_header->num_oids = unique_stat_info->unique_stats.num_oids;
-	  root_header->num_keys = unique_stat_info->unique_stats.num_keys;
+	  root_header->num_nulls = unique_stat_info->unique_stats.num_nulls & 0xffffffff;
+	  root_header->num_oids = unique_stat_info->unique_stats.num_oids & 0xffffffff;
+	  root_header->num_keys = unique_stat_info->unique_stats.num_keys & 0xffffffff;
+
+	  over += root_header->_64.num_nulls = unique_stat_info->unique_stats.num_nulls >> 32;
```

Comment:
It seems safe to check if the result of the operation "unique_stat_info->unique_stats.num_nulls >> 32" is greater than 2^10.

Reply:
unique_stat_info->unique_stats.num_nulls is 64 bit and overflown bits of 42 is not considered because the 42 bits is enough to represent the count reasonably.